src/iconv.cr

```diff
@@ -0,0 +1,39 @@
+@[Link("iconv")]
+lib LibIconv
+  type Iconv = Void*
+
+  fun open = iconv_open(tocode : LibC::Char*, fromcode : LibC::Char*) : Iconv
+  fun iconv(cd : Iconv, inbuf : LibC::Char**, inbytesleft : LibC::SizeT*, outbuf : LibC::Char**, outbytesleft : LibC::SizeT*) : LibC::SizeT
+  fun close = iconv_close(cd : Iconv) : LibC::Int
+end
+
+# :nodoc:
+struct Iconv
+  def initialize(from : String, to : String)
+    @iconv = LibIconv.open(to, from)
+    if @iconv.address == -1
+      raise ArgumentError.new("invalid encoding")
```

Comment:
Good idea. Here it's not possible because iconv doesn't tell you which one is invalid, but the ones that use this always have UTF-8 as one of them. I'll change it to catch the exception and re-raise with an appropriate message.